### PR TITLE
Testing incremental

### PR DESCRIPTION
**Pull request by griff122**
Current working version of the incremental. 

Looks like everything is updating correctly now. 

Some testing:
```
-- First run
dbadmin=> select * from testing.test_incremental  ;
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  1 |     4 | 2019-01-01 | 2019-02-14 19:47:09.841685+00
  2 |     3 | 2019-01-01 | 2019-02-14 19:47:09.841685+00
  3 |     5 | 2019-01-01 | 2019-02-14 19:47:09.841685+00

-- Second run with deltas added to source table:
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  3 |     5 | 2019-01-01 | 2019-02-14 19:47:54.460003+00
  1 |     6 | 2019-01-03 | 2019-02-14 19:48:00.404837+00
  2 |     7 | 2019-01-03 | 2019-02-14 19:48:00.404837+00
  4 |     8 | 2019-01-03 | 2019-02-14 19:48:00.404837+00

-- Third run with another round of deltas:
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  1 |     6 | 2019-01-03 | 2019-02-14 19:48:00.404837+00
  2 |     7 | 2019-01-03 | 2019-02-14 19:48:00.404837+00
  4 |     8 | 2019-01-03 | 2019-02-14 19:48:00.404837+00
  3 |     9 | 2019-01-04 | 2019-02-14 19:48:10.642081+00
  5 |    10 | 2019-01-04 | 2019-02-14 19:48:10.642081+00
  6 |    11 | 2019-01-04 | 2019-02-14 19:48:10.642081+00
```

In the second run there were two updated rows, and 1 new row. You can see the updated_on date and now timestamp for f1=3 does not change. 

In the third run, there were two new rows and one update. You can see the updated_on date and now timestamp for f1=(1,2,4) does not change, but the records where f1=(3,5,6) are updated.

### CODE REVIEW

**Comment by nabilihachem** [comment]
in the first delta f1=3 has a different timestamp than the original... Otherwise, all looks good.

File: dbt/include/global_project/macros/materializations/incremental/incremental.sql [hunk 1]
```diff
@@ -6,7 +6,7 @@
   from {{ target_relation }}
   where ({{ unique_key }}) in (
     select ({{ unique_key }})
-    from {{ tmp_relation.include(schema=False) }}
```
Comment (aderouineau) [inline dbt/include/global_project/macros/materializations/incremental/incremental.sql]: Should we be changing the global incremental.sql file?

Comment (griff122) [reply dbt/include/global_project/macros/materializations/incremental/incremental.sql]: I haven't been able to figure out how to use the `vertica_incremental.sql` file yet. I will need to dig in more to be able to see how to use it.

**Comment by griff122** [comment]
Good catch @nabilihachem ! I did not notice that. So there still seems to be an issue with the delete and insert that it is doing. I'll dig more into this. 

And @aderouineau - I don't believe deletes will work. That is a good case to consider.

Thanks!

**Comment by griff122** [comment]
Ok - so after going through the data a bunch of times, I'm not sure why that f1=3 timestamp changed during that run. It happened a couple more times, but I think I will chalk it up to user error. While running things today (and a bit more rested) - I'm not seeing the timestamp changing. 

Here is the new run:
```
dbadmin=> select * from testing.testtable4 ;
 f1 | delta | updated_on
----+-------+------------
  1 |     4 | 2019-01-01
  2 |     3 | 2019-01-01
  3 |     5 | 2019-01-01
(3 rows)

dbadmin=> select * from testing.test_incremental ;
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  1 |     4 | 2019-01-01 | 2019-02-19 13:00:02.340576+00
  2 |     3 | 2019-01-01 | 2019-02-19 13:00:02.340576+00
  3 |     5 | 2019-01-01 | 2019-02-19 13:00:02.340576+00
(3 rows)
```

Just the first run, so it looks good. Now for the next run: we are going to be updating `f1=(1,2)` and adding `f1=4` with an `updated_on=2019-01-03`:
```
dbadmin=> select * from testing.testtable4 ;
 f1 | delta | updated_on
----+-------+------------
  1 |     4 | 2019-01-01
  2 |     3 | 2019-01-01
  3 |     5 | 2019-01-01
  1 |     6 | 2019-01-03
  2 |     7 | 2019-01-03
  4 |     8 | 2019-01-03
(6 rows)

dbadmin=> select * from testing.test_incremental ;
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  3 |     5 | 2019-01-01 | 2019-02-19 13:00:02.340576+00
  1 |     6 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
  2 |     7 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
  4 |     8 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
(4 rows)
```

Now we can see that the `f1=3` has not changed. In the next test we added 2 new keys, `f1=(5,6)` where `updated_on=2019-01-04`:
```
dbadmin=> select * from testing.test_incremental ;
 f1 | delta | updated_on |              now
----+-------+------------+-------------------------------
  1 |     6 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
  2 |     7 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
  3 |     5 | 2019-01-01 | 2019-02-19 13:00:02.340576+00
  4 |     8 | 2019-01-03 | 2019-02-19 13:02:08.841086+00
  5 |    10 | 2019-01-04 | 2019-02-19 13:03:45.088095+00
  6 |    11 | 2019-01-04 | 2019-02-19 13:03:45.088095+00
(6 rows)
```

I think this is working now as expected. 

@aderouineau  - for the deletes, I think we need to establish correct rules for handling that type of event. I think the workflow should change based on if the source table is ephemeral or if it constantly being refreshed with the real source data. 

If this is OK for now, I would like to merge this in, and we can setup plans for how to handle the deletes. And we can setup actual expected tests to make sure that we are covering as many cases that we can.

**Pull request merged**